services/docker/docker-helpers.js

```diff
@@ -2,6 +2,24 @@
 import { NotFound } from '../index.js'
 const dockerBlue = '066da5'
 
+// Valid architecture values: https://golang.org/doc/install/source#environment (GOARCH)
+const validArchitectures = [
+  'amd64',
+  'arm',
+  'arm64',
+  's390x',
+  '386',
+  'ppc64',
+  'ppc64le',
+  'wasm',
+  'mips',
+  'mipsle',
+  'mips64',
+  'mips64le',
+  'riscv64',
+  386,
```

Comment:
Do we actually need `386` as a number as well as a string here?
If I look at an image which publishes a 386 version, for example:
https://registry.hub.docker.com/v2/repositories/library/debian/tags/latest
it appears in the JSON as a string:

```
"architecture": "386",
```

Do we have an example where the architecture is a number?

Reply:
OK. thanks for explaining. I can see what we're trying to achieve now. Here's my suggestion.

Lets define a shared Joi schema in `docker-helpers.js` that looks like this:

```js
const archSchema = Joi.alternatives([
  Joi.string().valid(
    'amd64',
    'arm',
    'arm64',
    's390x',
    '386',
    'ppc64',
    'ppc64le',
    'wasm',
    'mips',
    'mipsle',
    'mips64',
    'mips64le',
    'riscv64',
  ),
  Joi.number().valid(386).cast('string'),
])
```

That will accept all the normal architecture values and `386` as a number (but will cast it to a string so it will match the API data when we compare with `===` ).

Then in both `docker-size.service.js` and `docker-version.service.js` we can use that schema in our `queryParamSchema`, and in `docker-version.service.js` you can chain `.default('amd64')`

How does that sound?

Reply:
Good idea.
I checked for both `docker-size.service.js` and `docker-version.service.js`, and they still work correctly after the changes.

`docker-version.service.js` :

http://localhost:8080/docker/v/_/alpine?sort=date&arch=386
![image](https://user-images.githubusercontent.com/31891675/186658445-023fccf9-7e5f-4c6c-8d10-93e1719cdadc.png)

`docker-version.service.js` :
http://localhost:8080/docker/v/_/alpine?sort=date&arch=riscv64
![image](https://user-images.githubusercontent.com/31891675/186658327-46637585-1ca9-41c6-a637-8f7c3c736032.png)

http://localhost:8080/docker/v/_/alpine?sort=date&arch=notExistingArch
![image](https://user-images.githubusercontent.com/31891675/186658899-cab1bf58-2a91-4dfa-9876-43d3fbd57cc2.png)

---
`docker-size.service.js` :

http://localhost:8080/docker/image-size/_/mysql?arch=amd64
![image](https://user-images.githubusercontent.com/31891675/186658752-465f208e-e7e0-4fee-943c-d89aa9142c75.png)


http://localhost:8080/docker/image-size/_/mysql?arch=386
![image](https://user-images.githubusercontent.com/31891675/186658591-e71da409-9109-42cd-8740-a74ec934456c.png)

http://localhost:8080/docker/image-size/_/mysql?arch=notExistingArch
![image](https://user-images.githubusercontent.com/31891675/186658975-484a1e10-16a8-4382-ad95-3322a9868088.png)